app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,54 @@
+class VendorsController < ApplicationController
+
+  def index
+    @vendors = Vendor.order(vendor_name: :asc)
+    render :index
+  end
+
+  def show
+    @vendors = Vendor.where(id: params[:id])
+    render :single_vendor_view
+  end
+
+  def new
+    @new_vendor = Vendor.new
+    @market = Market.where(id: params[:id]).first
+  end
+
+  def create
+    @new_vendor = Vendor.new(new_vendor_create_params[:vendor])
+    @mark_id = @new_vendor.market_id
```

Comment:
Good note. Thank you. ...I don't even know why I did this, but I'll endeavor to use local variables in the future where appropriate.